### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by nevin75**

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        static float ValidFloatOrErrorCode(string input)
+        {
+            var isFloat = float.TryParse(input, out var number);
+            if (!isFloat)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+            else
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 1 above)
Comment (nevin75) [reply Src/BootCamp.Chapter/Lesson4.cs]: removed else

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            Console.WriteLine($"Height cannot be less than zero, but was {height}.");
+        }
+        public static void WeightErrorMessage(float weight)
+        {
+            Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+        }
+        public static char RequestUserChoiceToRepeatOrNot()
+        {
+            Console.WriteLine();
+            Console.Write("Would you like to repeat this program? (y or n)");
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]:
```suggestion
            Console.WriteLine("Would you like to repeat this program? (y or n)");
```

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            Console.WriteLine($"Height cannot be less than zero, but was {height}.");
+        }
+        public static void WeightErrorMessage(float weight)
+        {
+            Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+        }
+        public static char RequestUserChoiceToRepeatOrNot()
+        {
+            Console.WriteLine();
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]:
```suggestion

```

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            var isFloat = float.TryParse(input, out var number);
+            if (!isFloat)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+            return number;
+        }
+        public static bool ReportTrueIfUserInputIsEmpty(string input)
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: This function doesn't add anything to string.IsNullOrEmpty(input) and takes even more characters to write.
You should delete this one and use string.IsNullOrEmpty() instead.
If you think you have a good reason to do it that way feel free to explain your decision! :)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 4 above)
Comment (nevin75) [reply Src/BootCamp.Chapter/Lesson4.cs]: I think I was trying to keep logic abstracted, but it just adds complexity. I'll take it out. Thanks.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596&bid=16182610)

**Pull request closed without merging**